src/runtime/TemporalObject.cpp

```diff
@@ -1477,6 +1487,79 @@ TemporalTime::TemporalTime()
     , m_nanosecond(0)
 {
 }
+
+TemporalPlainMonthDay::TemporalPlainMonthDay(ExecutionState& state)
+    : TemporalPlainMonthDay(state, state.context()->globalObject()->objectPrototype())
+{
+}
+
+TemporalPlainMonthDay::TemporalPlainMonthDay(ExecutionState& state, Object* proto, int isoMonth, int isoDay, TemporalCalendar* calendar, int referenceISOYear)
+    : Temporal(state, proto)
+    , m_isoMonth(isoMonth)
+    , m_isoDay(isoDay)
+    , m_calendar(calendar)
+    , m_referenceISOYear(referenceISOYear)
+{
+}
+
+Value TemporalPlainMonthDay::createTemporalMonthDay(ExecutionState& state, int isoMonth, int isoDay, TemporalCalendar* calendar, int referenceISOYear, Optional<Object*> newTarget)
+{
+    if (!TemporalPlainDate::isValidISODate(state, referenceISOYear, isoMonth, isoDay)) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "Invalid date");
+    }
+
+    if (!newTarget.hasValue()) {
+        newTarget = state.resolveCallee();
+    }
+
+    Object* proto = Object::getPrototypeFromConstructor(state, newTarget.value(), [](ExecutionState& state, Context* constructorRealm) -> Object* {
+        return constructorRealm->globalObject()->temporal()->asTemporalObject()->getTemporalPlainMonthDayPrototype();
+    });
+
+
+    return new TemporalPlainMonthDay(state, proto, isoMonth, isoDay, calendar, referenceISOYear);
+}
+
+Value TemporalPlainMonthDay::toTemporalMonthDay(ExecutionState& state, const Value& item, const Value& options)
+{
+    ASSERT(options.isObject() || options.isUndefined());
+    int referenceISOYear = 1972;
+
+    if (item.isObject()) {
```

Comment:
Since the else is shorter, I would swap these to reduce indentation. Also add a newlines after the ifs to look better.